src/Domain/EmbeddingMaps/Rotation.hpp

```diff
@@ -0,0 +1,177 @@
+  explicit Rotation(double rotation_angle);
+  Rotation() = default;
+  ~Rotation() override = default;
+  Rotation(const Rotation&) = delete;
+  Rotation& operator=(const Rotation&) = delete;
+  Rotation(Rotation&&) noexcept = default;  // NOLINT
+  Rotation& operator=(Rotation&&) = delete;
+
+  std::unique_ptr<EmbeddingMap<2, 2>> get_clone() const override {
+    return std::make_unique<Rotation<2>>(rotation_angle_);
+  }
+
+  Point<2, Frame::Grid> operator()(
+      const Point<2, Frame::Logical>& xi) const override;
+
+  Point<2, Frame::Logical> inverse(
+      const Point<2, Frame::Grid>& x) const override;
+
+  double jacobian(const Point<2, Frame::Logical>& /*xi*/, size_t ud,
+                  size_t ld) const override {
+    ASSERT(0 == ld || 1 == ld, "ld = " << ld);
+    ASSERT(0 == ud || 1 == ud, "ud = " << ud);
+    return rotation_matrix_.get(ud, ld);
+  }
+
+  double inv_jacobian(const Point<2, Frame::Logical>& /*xi*/, size_t ud,
+                      size_t ld) const override {
+    ASSERT(0 == ld || 1 == ld, "ld = " << ld);
+    ASSERT(0 == ud || 1 == ud, "ud = " << ud);
+    // Recall inverse(rotation_matrix_) = transpose(rotation_matrix_)
+    return rotation_matrix_.get(ld, ud);
+  }
+
+  WRAPPED_PUPable_decl_base_template(SINGLE_ARG(EmbeddingMap<2, 2>),  // NOLINT
+                                     Rotation);                       // NOLINT
+
+  explicit Rotation(CkMigrateMessage* /* m */)
+      : rotation_angle_(std::numeric_limits<double>::signaling_NaN()),
+        rotation_matrix_() {}
+  void pup(PUP::er& p) override {  // NOLINT
+    p | rotation_angle_;
+    p | rotation_matrix_;
+  }
+
+ private:
+  double rotation_angle_{std::numeric_limits<double>::signaling_NaN()};
+  tnsr::ij<double, 2, Frame::Grid> rotation_matrix_{
+      std::numeric_limits<double>::signaling_NaN()};
+};
+
+/// Spatial rotation in three dimensions using Euler angles
+///
+/// Rotation angles should be specified in degrees.
+/// First rotation \f$\alpha\f$ is about z axis.
+/// Second rotation \f$\beta\f$ is about rotated y axis.
+/// Third rotation \f$\gamma\f$ is about rotated z axis.
+/// These rotations are of the \f$(\xi,\eta,\zeta)\f$ coordinate system with
+/// respect
+/// to the grid coordinates \f$(x,y,z)\f$.
+///
+/// The formula for the mapping is:
+/*!\f{eqnarray*}
+    x &=& \xi (\cos\gamma \cos\beta \cos\alpha - \sin\gamma \sin\alpha)
+        + \eta (-\sin\gamma \cos\beta \cos\alpha - \cos\gamma \sin\alpha)
+        + \zeta \sin\beta \cos\alpha
+\\  y &=& \xi (\cos\gamma \cos\beta \sin\alpha + \sin\gamma \cos\alpha)
+        + \eta (-\sin\gamma \cos\beta \sin\alpha + \cos\gamma \cos\alpha)
+        + \zeta \sin\beta \sin\alpha
+\\  z &=& -\xi \cos\gamma \sin\beta + \eta \sin\gamma \sin\beta
+        +  \zeta \cos\beta
+\f}
+*/
+template <>
+class Rotation<3> : public EmbeddingMap<3, 3> {
+ public:
+  /// Constructor.
+  ///
+  /// \param rotation_about_z the angle \f$\alpha\f$ (in radians).
```

Comment:
I don't think so as this is a static rotation.  The rotation matrix will be the same whether you use Euler angles or quaternions.